src/main/java/seedu/duke/storage/TaskDecoder.java

```diff
@@ -0,0 +1,85 @@
+package seedu.duke.storage;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import seedu.duke.common.Messages;
+import seedu.duke.exception.StorageOperationException;
+import seedu.duke.task.Deadline;
+import seedu.duke.task.Event;
+import seedu.duke.task.Task;
+import seedu.duke.task.Todo;
+
+
+/**
+ * The TaskDecoder class implements a method that would decode string and parse it into Task object
+ */
+public class TaskDecoder {
```

Comment:
Hey BaiZhou! I like the way you implemented this TaskDecoder for your save/load purposes! Clean and neat which really made it easy to understand